pkg/kapp/resourcesmisc/wait_rule_contract_v1.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package resourcesmisc
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+	cmdtpl "github.com/k14s/ytt/pkg/cmd/template"
+	"github.com/k14s/ytt/pkg/cmd/ui"
+	"github.com/k14s/ytt/pkg/files"
+)
+
+type WaitRuleContractV1 struct {
+	ResourceMatcher ctlres.ResourceMatcher
+	Starlark        string
+}
+
+type waitRuleContractV1Result struct {
+	Result waitRuleContractV1ResultDetails
+}
+
+type waitRuleContractV1ResultDetails struct {
+	Done       bool   `json:"done"`
+	Successful bool   `json:"successful"`
+	Message    string `json:"message"`
+}
+
+func (t WaitRuleContractV1) applyYttWaitRule(res ctlres.Resource) (*waitRuleContractV1ResultDetails, error) {
```

Comment:
applyYttWaitRule => Apply. should be public since we expect other classes to use it.

Reply:
Making it public also means that the return type should also be public otherwise lint check is getting failed. 

Made `WaitRuleContractV1ResultDetails` public.